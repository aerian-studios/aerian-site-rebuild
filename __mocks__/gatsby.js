"use strict";
import React from "react";
const gatsby = jest.genMockFromModule("gatsby");
gatsby.graphql = jest.fn();
gatsby.Link = ({ to, activeClassName, activeStyle, isActive, ...props }) => (
    <a
        href={to}
        activeclassname={activeClassName}
        activestyle={activeStyle}
        isactive={isActive}
        {...props}
    />
);

module.exports = gatsby;
