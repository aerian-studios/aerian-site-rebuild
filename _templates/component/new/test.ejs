---
to: src/components/<%= h.inflection.undasherize(name) %>/<%= h.inflection.undasherize(name) %>Test.tsx
---
<% const comp = h.inflection.undasherize(name) -%>
/// <reference types="@types/jest" />
import * as React from "react";

import * as renderer from "react-test-renderer";

import { <%= comp %> } from "./index";


describe("<%= comp %>", () => (
    it("renders correctly", () => {
        const tree = renderer
            .create(<<%= comp %> className="myClass" />
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    })
));

