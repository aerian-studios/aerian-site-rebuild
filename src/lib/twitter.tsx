import React from "react";

import { Tweet } from "../types/data";

export const parseTweet = (tweet: Tweet) => {
    const { urls } = tweet.entities;
    if (!urls || !urls.length) {
        return tweet.full_text;
    }

    const text = tweet.full_text;

    const elements: React.ReactNode[] = [text.substring(0, urls[0].indices[0])];
    for (let i = 0; i < urls.length; i++) {
        const url = urls[i];
        elements.push(
            <a
                key={url.url}
                href={url.expanded_url}
                target="_blank"
                rel="noreferrer noopener nofollow"
            >
                {url.display_url}
            </a>
        );
        const nextIndex =
            i === urls.length - 1 ? text.length : urls[i + 1].indices[0];
        elements.push(text.substring(url.indices[1], nextIndex));
    }

    return <>{elements}</>;
};
