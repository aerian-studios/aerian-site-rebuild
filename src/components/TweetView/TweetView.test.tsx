import * as React from "react";

import renderer from "react-test-renderer";

import { TweetView } from ".";

import { mockTweet } from "../../lib/twitter.test";
import { useStaticQuery } from "gatsby";

describe("TweetView", () => {
    beforeEach(() => {
        useStaticQuery.mockReturnValueOnce({ tweet: mockTweet });
    });
    it("renders correctly", () => {
        const tree = renderer
            .create(<TweetView className="myClass" />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
