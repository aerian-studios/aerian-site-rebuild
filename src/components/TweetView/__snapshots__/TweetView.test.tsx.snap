// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TweetView renders correctly 1`] = `
<div
  className="myClass"
>
  <div>
    MOCK!: proud that our partners over at @CRbuildpeace are being featured by our friends at @BBCRadio4 this Sunday. What's more, we've recently just launched their new site - 
    <a
      href="http://www.c-r.org"
      rel="noreferrer noopener nofollow"
      target="_blank"
    >
      c-r.org
    </a>
    . Check it out! 
    <a
      href="https://twitter.com/CRbuildpeace/status/1164462963790667776"
      rel="noreferrer noopener nofollow"
      target="_blank"
    >
      twitter.com/CRbuildpeace/sâ€¦
    </a>
    
  </div>
  <a
    href="https://twitter.com/aerianstudios"
  >
    <img
      alt="Aerian"
      src="https://pbs.twimg.com/profile_images/378800000156917292/cba17ed82c5a99f3b91e805534b2d10f_normal.png"
    />
    <p>
      <strong>
        Tweeted by
      </strong>
      Aerian
    </p>
  </a>
</div>
`;
